### ISSUE TITLE
linter: eslint-plugin-jsdoc(require-returns) false positive if anything before @returns uses one single quote

### ISSUE THREAD

**Issue by yunarch** [oxc-project/oxc#13776]
### What version of Oxlint are you using?

1.15.0

### What command did you run?

oxlint

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "$schema": "https://raw.githubusercontent.com/oxc-project/oxc/main/npm/oxlint/configuration_schema.json",
  "plugins": ["jsdoc"],
  "rules": {
    "jsdoc/require-returns": "error"
  }
}
```

### What happened?

It looks like the JSDoc parser/checker breaks when a single quote (') is used inside a comment.

For example, this works fine:
```ts
/**
 * Test description.
 * 
 * @returns false
 */
function test() {
  return false;
}
```

But this fails and it shouldn't.
```ts
/**
 * Test description, this shouldn't lint error.
 * 
 * @returns false
 */
function test() {
  return false;
}
```

If I remove the single quote or replace it with another character (for example, a typographic apostrophe `’`), the error goes away. It also disappears if I use two single quotes (e.g., `shouldn''t`). However, these are not acceptable solutions. A single quote (`'`) should be valid grammar character for English contractions and should be supported correctly inside JSDoc comments.

### Expected Behavior

- JSDoc should allow normal English contractions like `shouldn't` without breaking the parser.

### Actual Behavior

- Using a single quote (or a single double quote) inside a JSDoc block causes the parser/checker to lint an error:
```ts
Missing JSDoc `@returns` declaration for function.
help: Add `@returns` tag to the JSDoc comment.oxc[eslint-plugin-jsdoc(require-returns)](https://oxc.rs/docs/guide/usage/linter/rules/jsdoc/require-returns.html)
```

> It only breaks if the single (') or (") is used before the `@returns` tag, inside that tag or after works.

**Comment by yunarch** [comment]
A small clarification: the issue appears when single quotes (`'`) or double quotes (`"`) are used **unpaired** (maybe other characters as well). If they are used in pairs (opening and closing), it works. However, pairing should not be required for single quotes, as explained in the issue. It might make sense for double quotes, but not for single quotes.

**Comment by yunarch** [comment]
@Boshen @camc314 
I’ve tested this in the latest version `1.23.0`, and the issue is still reproducible. Can we reopen the issue or should I create a new one?

**Comment by camc314** [comment]
I'm planning to do a new release today (1.24.0), do you mind checking once that is out, and reporting back?

If it's still an issue, please open a new issue. Thanks

**Comment by yunarch** [comment]
@camc314 Looks like is working good in the latest release (`v1.24.0`). Thank you so much 😄

**Comment by camc314** [comment]
perfect, thank you for checking!

**Comment by eikowagenknecht** [comment]
I stumbled across a very close variation of this bug today, seems that braces are affected as well if they are not in proper pairs.

This also leads to a false positive:
```
/**
 * Escaped characters: )
 * @returns The escaped text
 */
export function escapeCode(text: string): string {
  return ...
}
```

Same with `\)` or `())`. Also for other brace types like `]` or `}`.
While these work: `()`, `\(\)`
